src/coord_seq.rs

```diff
@@ -928,3 +937,475 @@ impl<'a> ContextHandling for CoordSeq<'a> {
         Arc::clone(&self.context)
     }
 }
+
+impl<'a, 'b> ConstCoordSeq<'a, 'b> {
+    pub(crate) unsafe fn new_from_raw(
+        ptr: *const GEOSCoordSequence,
+        original: &'b Geometry<'a>,
+        size: u32,
+        dims: u32,
+        caller: &str,
+    ) -> GResult<ConstCoordSeq<'a, 'b>> {
+        if ptr.is_null() {
+            let extra = if let Some(x) = original.context.get_last_error() {
+                format!("\nLast error: {x}")
+            } else {
+                String::new()
+            };
+            return Err(Error::NoConstructionFromNullPtr(format!(
+                "CoordSeq::{caller}{extra}",
+            )));
```

Comment:
You're calling two `format!` here, which isn't great. Could you instead have `format!`s directly into the `let Some(x) else` please?